### ISSUE TITLE
Build failure with fedora 38 and 37

### ISSUE THREAD

**Issue by raveit65** [zhuyaliang/group-service#8]
Hi,
During the mass rebuild for f38 we got this build failure.
```
Run-time dependency systemd found: NO (tried pkgconfig and cmake)
meson.build:37:2: ERROR: Assert failed: systemd required but not found, please provide a valid systemd user unit dir or disable     it
A full log can be found at /builddir/build/BUILD/group-service-1.3.0/redhat-linux-build/meson-logs/meson-log.txt
error: Bad exit status from /var/tmp/rpm-tmp.U5jOHv (%build)
    Bad exit status from /var/tmp/rpm-tmp.U5jOHv (%build)
```
build log at https://kojipkgs.fedoraproject.org//work/tasks/1585/96351585/build.log
https://koji.fedoraproject.org/koji/taskinfo?taskID=96351585

Build is for arch64, but we have same results with x86_64 arch. And the issue still exists for f37.
Issue is triggered by this macro `%{?systemd_requires}` in rpm spec file https://src.fedoraproject.org/rpms/group-service/blob/rawhide/f/group-service.spec#_27
This macro is needed to install the systemd-unit-file.
And i think this part in meson.build file caused the build failure.
```
systemd_system_unit_dir = get_option('systemdsystemunitdir')
install_systemd_unit_dir = (systemd_system_unit_dir != 'no')
enable_systemd = get_option('systemd')
enable_elogind = get_option('elogind')
```
group-service builds fine with f36, so i think something with meson was changed.

**Comment by zhuyaliang** [comment]
This should not be the problem of ```group-service``` itself,missing ```systemd``` dependency in compilation environment,
### This is the correct installation method [root.log](https://kojipkgs.fedoraproject.org//packages/group-service/1.3.0/7.fc38/data/logs/aarch64/root.log)
```
DEBUG util.py:445:  Installing:
DEBUG util.py:445:   cmake                     aarch64   3.25.2-1.fc38                build   6.5 M
DEBUG util.py:445:   dbus-devel                aarch64   1:1.14.4-2.fc38              build    35 k
DEBUG util.py:445:   gcc                       aarch64   13.0.1-0.1.fc38.1            build    31 M
DEBUG util.py:445:   gettext                   aarch64   0.21.1-1.fc38                build   1.0 M
DEBUG util.py:445:   glib2-devel               aarch64   2.74.1-3.fc38                build   517 k
DEBUG util.py:445:   libxcrypt-devel           aarch64   4.4.33-6.fc38                build    30 k
DEBUG util.py:445:   meson                     noarch    1.0.0-3.fc38                 build   1.9 M
DEBUG util.py:445:   polkit-devel              aarch64   122-3.fc38                   build    40 k
DEBUG util.py:445:   systemd                   aarch64   252.4-4.fc38                 build   4.1 M
DEBUG util.py:445:   systemd-rpm-macros        noarch    252.4-4.fc38                 build    31 k
```

### This is the wrong installation [root.log](https://kojipkgs.fedoraproject.org//work/tasks/1585/96351585/root.log)
```
DEBUG util.py:445:  Installing:
DEBUG util.py:445:   cmake                     aarch64   3.25.1-3.fc38                build   6.3 M
DEBUG util.py:445:   dbus-devel                aarch64   1:1.14.4-1.fc38              build    34 k
DEBUG util.py:445:   gcc                       aarch64   13.0.1-0.1.fc38              build    31 M
DEBUG util.py:445:   gettext                   aarch64   0.21.1-1.fc38                build   1.0 M
DEBUG util.py:445:   glib2-devel               aarch64   2.74.1-2.fc38                build   518 k
DEBUG util.py:445:   libxcrypt-devel           aarch64   4.4.33-5.fc38                build    30 k
DEBUG util.py:445:   meson                     noarch    1.0.0-1.fc38                 build   1.9 M
DEBUG util.py:445:   polkit-devel              aarch64   122-2.fc38                   build    40 k
DEBUG util.py:445:   systemd-devel             aarch64   252.4-598.fc38               build   468 k
DEBUG util.py:445:   systemd-rpm-macros        noarch    252.4-598.fc38               build    31 k
```

We can fix this problem in the ```group-service.spec``` file
```
 | BuildRequires:  systemd
 | BuildRequires:  systemd-devel

```

**Comment by zhuyaliang** [comment]
[group-service](https://github.com/zhuyaliang/group-service/commit/94a1e3885f715590d99ef91bb093d6b52db0294b) and [user-admin](https://github.com/zhuyaliang/user-admin/commit/0a5dc793ce41cacbcca93c7e59c3cd13da31741b) have been updated

**Comment by raveit65** [comment]
Thanks that works for fedora. The group-service- 1.4.0 update works fine for fedora but not for rhel9 (epel9 branch), see https://koji.fedoraproject.org/koji/taskinfo?taskID=96703784
https://kojipkgs.fedoraproject.org//work/tasks/3784/96703784/build.log
```
/builddir/build/BUILD/group-service-1.4.0/redhat-linux-build/../src/util.c:203: undefined reference to `g_spawn_check_wait_status'
collect2: error: ld returned 1 exit status
```
Looks like the `GLIB_CHECK_VERSION(2, 33, 12)` doesn't work
Glib version is:
glib2-devel-2.68.4-5.el9.x86_64

**Comment by zhuyaliang** [comment]
@raveit65 The problem has been fixed [ca7ef9a](https://github.com/zhuyaliang/group-service/commit/ca7ef9adf5772733b4e478facd4308ec461b8db0)

**Comment by raveit65** [comment]
Works like expected now.
https://koji.fedoraproject.org/koji/taskinfo?taskID=96889268
https://koji.fedoraproject.org/koji/taskinfo?taskID=96889976
Thanks
